src/Search/Extensions/IndexableExtension.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace SilverStripe\FullTextSearch\Search\Extensions;
+
+use SilverStripe\ORM\DataExtension;
+
+/**
+ * Checks if a DataObject is publically viewable thus able to be added or retrieved from a publically searchable index
+ */
+class IndexableExtension extends DataExtension
+{
+
+    protected $hasCheckedIsIndexable = false;
+
+    protected $canBeIndexed = false;
+
+    /**
+     * @return bool
+     */
+    public function isIndexable()
+    {
+        if ($this->owner->hasCheckedIsIndexable) {
```

Comment:
You could also do this as a tri-state, and check for `canBeIndexed === NULL`. The advantage of that is that it's easier to perform test setups mainly. And you don't get into awkward naming disparities like "can be indexed" vs "has checked is indexable" which are correlated, but not in naming.

Reply:
Changed to a tri-state.  I used const instead of nullable boolean for a bit more clarify and less likely to make mistakes with `if (!$this->owner->indexedState)`